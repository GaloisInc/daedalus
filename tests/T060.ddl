def Main = {
  SetStream (arrayStream [1,2]);
  ex1 = case UInt8 of {
          1 -> true;
          2 -> false;
        };
}

