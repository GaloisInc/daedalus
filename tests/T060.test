T060.ddl
--json
-i
