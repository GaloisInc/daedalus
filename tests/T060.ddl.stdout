module T060
 
--- Imports:
 
--- Type defs:
type T060.Main = { ex1: bool
                 }
 
--- Rules:
 
T060.Main : Grammar T060.Main =
  do SetStream (arrayStream "array" [1,
                                     2])
     (ex1 : bool) <- do (_2 : uint 8) <- Match UInt8
                        case _2 is
                          { 1 -> {- case branch  1 -} pure true
                          ; 2 -> {- case branch  2 -} pure false
                          }
     pure {ex1 = ex1}
 
T060._Main : Grammar {} =
  do SetStream (arrayStream "array" [1,
                                     2])
     (_2 : uint 8) <- Match UInt8
     case _2 is
       { 1 -> {- case branch  1 -} pure {}
       ; 2 -> {- case branch  2 -} pure {}
       }
